group.go

```diff
@@ -864,3 +864,54 @@ func (cli *Client) parseGroupNotification(node *waBinary.Node) (interface{}, err
 		return groupChange, nil
 	}
 }
+
+// GroupJoinApprovalMode sets the group join approval mode to 'on' or 'off'.
+func (cli *Client) GroupJoinApprovalMode(jid types.JID, mode string) error {
+	if mode != "on" && mode != "off" {
+		return errors.New("invalid mode, must be 'on' or 'off'")
+	}
+
+	content := waBinary.Node{
+		Tag: "membership_approval_mode",
+		Content: []waBinary.Node{
+			{
+				Tag:   "group_join",
+				Attrs: waBinary.Attrs{"state": mode},
+			},
+		},
+	}
+
+	_, err := cli.sendGroupIQ(context.TODO(), iqSet, jid, content)
+	return err
+}
+
+// GroupMemberAddMode sets the group member add mode to 'admin_add' or 'all_member_add'.
+func (cli *Client) GroupMemberAddMode(jid types.JID, mode string) error {
+	if mode != "admin_add" && mode != "all_member_add" {
```

Comment:
This needs to use the existing type rather than a string https://pkg.go.dev/go.mau.fi/whatsmeow/types#GroupMemberAddMode

Reply:
it's done.